csharp/private/actions/assembly.bzl

```diff
@@ -53,15 +53,17 @@ def AssemblyAction(
         langversion,
         resources,
         srcs,
+        out,
         target,
         target_framework,
         toolchain):
 
+    out_file_name = name if out == None else out
```

Comment:
nit: I think `out or name` name would work/be idiomatic

Reply:
Have moved from `None` to `""`, and I'm not sure how idiomatic it would be to use that construct for an empty string.